### DISCUSSION TITLE
persist with sessionStorage

### DISCUSSION THREAD

**Discussion by leesei** [alpinejs/alpine#2927] [3. Feature ideas]
From persist's source the storage is hardcoded to localStorage.
https://github.com/alpinejs/alpine/blob/main/packages/persist/src/index.js#L4

Supporting sessionStorage is as easy as an if-else switch.
However it seems we cannot pass parameters to plugin constructor, is it right?
Any comments?

**Comment by SimoTod** [comment]
It's not hardcoded, that line is just the default value. See docs: https://alpinejs.dev/plugins/persist#custom-storage

**Comment by leesei** [comment]
I see. But how does `.using()` gets applied to the `$persist` instance?
Is it related to the internal `interceptor()` API?

**Reply by SimoTod** [reply]
This is the relevant line of code
https://github.com/alpinejs/alpine/blob/main/packages/persist/src/index.js#L26

**Reply by leesei** [reply]
Thanks. New to Alpine, don't know what to look for.
But it actually seems highly extensible and intuitive :+1: .
